drivers/bridge/setup_ip_tables.go

```diff
@@ -36,10 +37,26 @@ func setupIPChains(config *configuration) (*iptables.ChainInfo, *iptables.ChainI
 
 	filterChain, err := iptables.NewChain(DockerChain, iptables.Filter, hairpinMode)
 	if err != nil {
-		return nil, nil, fmt.Errorf("Failed to create FILTER chain: %s", err.Error())
+		return nil, nil, nil, fmt.Errorf("Failed to create FILTER chain: %v", err)
+	}
+	defer func() {
+		if err != nil {
+			if err := iptables.RemoveExistingChain(DockerChain, iptables.Filter); err != nil {
+				logrus.Warnf("Failed on removing iptables FILTER chain on cleanup: %v", err)
+			}
+		}
+	}()
+
+	isolationChain, err := iptables.NewChain(IsolationChain, iptables.Filter, hairpinMode)
```

Comment:
You can pass false. It is only applicable to `Nat` table so it is optional argument. The current way creates confusion for anybody reading the code who are going to think that hairpin mode means something to filter table as well. You can also remove it from the DOCKER filter table chain also if you would like consistency.